src/filter/ahash/aes_hash.rs

```diff
@@ -0,0 +1,424 @@
+// Original work: Copyright (c) 2018 Tom Kaitchuck
+// Distributed under MIT license
+// Taken from aHash version 0.7.4 (commit ffa04fcb81f39755f636c75c9b7aa06533c0ae75)
+
+use super::convert::*;
+#[cfg(feature = "specialize")]
+use super::fallback_hash::MULTIPLE;
+use super::operations::*;
+use super::PI;
+use core::hash::Hasher;
+
+/// A `Hasher` for hashing an arbitrary stream of bytes.
+///
+/// Instances of [`AHasher`] represent state that is updated while hashing data.
+///
+/// Each method updates the internal state based on the new data provided. Once
+/// all of the data has been provided, the resulting hash can be obtained by calling
+/// `finish()`
+///
+/// [Clone] is also provided in case you wish to calculate hashes for two different items that
+/// start with the same data.
+///
+#[derive(Debug, Clone)]
+pub struct AHasher {
```

Comment:
If this hardware-accelerated hasher produces a different hash value than what is programmed in our tests, then it is better to remove it completely and leave only the software version